sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/CurrentUser.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.codegen.CodegenFallback
+import org.apache.spark.sql.types.{DataType, StringType}
+
+/**
+ * Return the user executing the current query.
+ *
+ * There is no code generation since this expression should get constant folded by the optimizer.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_() - Returns the current user executing the query.
+  """,
+  since = "2.3.0")
+case class CurrentUser() extends LeafExpression with CodegenFallback {
+  override def foldable: Boolean = true
+  override def nullable: Boolean = false
+  override def dataType: DataType = StringType
+
+  /** Returns the result of evaluating this expression on a given input Row */
+  override def eval(input: InternalRow): Any = {
+    SparkContext.getActive.map(_.sparkUser).getOrElse("")
```

Comment:
This is from Spark Context. All the sessions share the same user names. I am afraid this value is misleading to our end users. cc @rxin

Reply:
This is actually the user which executes the query. I agree that may be misleading that a user can connect to STS with his credentials and then the queries are run with a superuser, but this is how Spark works at the moment and the user which actually runs the query is the superuser. If we don't think to STS, in the other use cases the user which runs a query is the user who started the session.

Reply:
Sorry, I do not think we should support it until we provide session-specific user management.

Reply:
ok, thanks. Then is there any plan to support session-specific user management? If so, can I help somehow? Thanks.

Reply:
So far, it is not in our plan.